script/tool/lib/src/publish_plugin_command.dart

```diff
@@ -120,7 +211,50 @@ class PublishPluginCommand extends PluginCommand {
           remoteUrl: remoteUrl,
           shouldPushTag: shouldPushTag);
     }
-    await _finishSuccesfully();
+    _print('Release ${packageDir.basename} successful.');
+  }
+
+  // Returns `true` if needs to release the version, `false` if needs to skip
+  Future<bool> _checkNeedsRelease({
+    @required String pubspecPath,
+    @required File pubspecFile,
+    @required GitVersionFinder gitVersionFinder,
+    @required List<String> existingTags,
+  }) async {
+    final Pubspec pubspec = Pubspec.parse(pubspecFile.readAsStringSync());
+    if (pubspec.publishTo == 'none') {
+      return false;
+    }
+
+    final Version headVersion =
+        await gitVersionFinder.getPackageVersion(pubspecPath, gitRef: 'HEAD');
+    if (headVersion == null) {
+      printErrorAndExit(
+          errorMessage: 'No version found. A package that '
+              'intentionally has no version should be marked '
+              '"publish_to: none".');
+    }
+
+    if (pubspec.name == null) {
+      printErrorAndExit(errorMessage: 'Fatal: Package name is null.');
+    }
+    // Get latest tagged version and compare with the current version.
+    // TODO(cyanglaz): Check latest version of the package on pub instead of git
+    // https://github.com/flutter/flutter/issues/81047
+    final String latestTag = existingTags.isNotEmpty
+        ? existingTags
+            .firstWhere((String tag) => tag.split('-v').first == pubspec.name)
+        : '';
```

Comment:
You can use `orElse` here instead of checking `isNotEmpty`.

In fact, I think without that change the current logic is wrong, since just because the set of tags isn't empty doesn't mean that this particular package has been released before, so it seems like this would throw as soon as we had a new plugin. (Which suggests there's a test case missing.)